llrt_core/src/modules/encoding/text_decoder.rs

```diff
@@ -64,24 +63,31 @@ impl<'js> TextDecoder {
     }
 
     pub fn decode(&self, ctx: Ctx<'js>, buffer: Value<'js>) -> Result<String> {
-        let bytes = get_bytes(&ctx, buffer)?;
-
-        let decoder = Encoding::for_label(self.encoding.as_bytes()).unwrap();
+        let bytes = if buffer.is_undefined() {
+            vec![]
+        } else {
+            get_bytes(&ctx, buffer)?
+        };
```

Comment:
Should this be added into `get_bytes`? Is it expected behavior for other places using bytes from a buffer, typed array or or string?

Reply:
Good question. I'll look where else it's used. Once we implement the `stream` option, I think this behavior changes even more, but I haven't read the spec yet; but there are wpt test cases around flushing the streaming buffer when `buffer` is `undefined` at the end of a stream.